src/components/CustomTextInput.tsx

```diff
@@ -0,0 +1,123 @@
+  border-radius: 2px;
+  height: ${props => (props.lines > 1 ? props.lines * LINE_HEIGHT : MIN_HEIGHT)}px;
+  flex-direction: row;
+  margin-bottom: ${props => (props.showErrorValidation ? 0 : props.theme.spacings.xs)};
+`
+
+const IconContainer = styled.View<{ multiLine: boolean }>`
+  align-self: ${props => (props.multiLine ? 'flex-start' : 'center')};
+  padding: ${props => props.theme.spacings.xs} 0;
+`
+
+const ErrorContainer = styled.View`
+  min-height: ${props => props.theme.spacings.lg};
+`
+
+interface CustomTextInputProps extends TextInputProps {
+  value: string
+  onChangeText: (value: string) => void
+  /** calculates the height for textArea */
+  lines?: number
+  /** shows clear indicator if set & input set */
+  clearable?: boolean
+  rightContainer?: ReactElement
+  /** Shows error message below the text input. Empty string provides a placeholder */
+  errorMessage?: string
+  /** Custom border color can be defined */
+  customBorderColor?: string
+}
+
+// depending on error and focus the border color will be defined
+const getBorderColor = (hasErrorMessage: boolean, isFocused: boolean): string => {
+  if (hasErrorMessage) {
+    return theme.colors.incorrect
+  }
+  return isFocused ? theme.colors.primary : theme.colors.textSecondary
+}
+
+const CustomTextInput: React.FC<CustomTextInputProps> = ({
+  value,
+  onChangeText,
+  lines = 1,
+  clearable = false,
+  placeholder,
+  textContentType,
+  rightContainer,
+  errorMessage,
+  editable = true,
+  onSubmitEditing,
+  customBorderColor
+}: CustomTextInputProps): ReactElement => {
+  const theme = useTheme()
+  const [isFocused, setIsFocused] = useState<boolean>(false)
+  const hasErrorMessage = !!(errorMessage && errorMessage.length > 0)
+  const showErrorValidation = errorMessage !== undefined
+  const multiLine = lines > 1
+
+  return (
+    <>
+      <TextInputContainer
+        lines={lines}
+        borderColor={customBorderColor ?? getBorderColor(hasErrorMessage, isFocused)}
+        showErrorValidation={showErrorValidation}>
+        <StyledTextInput
+          onFocus={() => setIsFocused(true)}
+          onBlur={() => setIsFocused(false)}
+          textContentType={textContentType}
+          value={value}
+          onChangeText={onChangeText}
+          placeholder={placeholder}
+          multiline={multiLine}
+          placeholderTextColor={theme.colors.placeholder}
+          editable={editable}
+          onSubmitEditing={onSubmitEditing}
+          textAlignVertical={multiLine ? 'top' : 'center'}
+        />
+        <IconContainer multiLine={multiLine}>
+          {clearable && value.length > 0 ? (
+            <ClearContainer onPress={() => onChangeText('')} testID='clearInput'>
+              <CloseIcon width={wp('6%')} height={wp('6%')} />
+            </ClearContainer>
+          ) : (
+            rightContainer
+          )}
+        </IconContainer>
+      </TextInputContainer>
+      {showErrorValidation && (
```

Comment:
hm i think a text input base component should have some kind of field validation (and error message) because we may need it in the future when the user sends more info to the cms. I can refactor it to be more generic for Write-Exercise, i just didn't want to make it too complicated